### PR TITLE
Feature/play against computer

### PR DESCRIPTION
**Pull request by hyperxtend**

### CODE REVIEW

File: src/pages/landing-page/component.test.jsx [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import React from 'react';
+import { shallow } from 'enzyme';
+
+import LandingPage from './component';
+
+describe('<LandingPage/>', () => {
+  const component = shallow(<LandingPage />);
+
+  it('checks description on landing page', () => {
+    const descriptionSubHeader = component.find('h3').children().first().text();
```
Comment (strange-developer) [inline src/pages/landing-page/component.test.jsx]: Prefer QA tags

File: src/pages/landing-page/component.test.jsx [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import React from 'react';
+import { shallow } from 'enzyme';
+
+import LandingPage from './component';
+
+describe('<LandingPage/>', () => {
+  const component = shallow(<LandingPage />);
```
Comment (strange-developer) [inline src/pages/landing-page/component.test.jsx]: Change this variable name to wrapper

File: src/pages/landing-page/component.test.jsx [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+  it('checks description on landing page', () => {
+    const descriptionSubHeader = component.find('h3').children().first().text();
+    const descriptionContent =
+      'This adaptation of the classic Tic-Tac-Toe where players are able to go through their previous moves, play against a friend or against the computer';
+    expect(descriptionSubHeader).toEqual(descriptionContent);
+  });
+
+  it('checks sub-header under game description', () => {
+    const descriptionSubHeader = component.find('h5').children().first().text();
```
Comment (strange-developer) [inline src/pages/landing-page/component.test.jsx]: Prefer QA tags

File: src/pages/landing-page/menu-selection/component.test.jsx [hunk 4]
```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+import { shallow } from 'enzyme';
+
+import MenuSelection from './component';
+
+describe('<MenuSelection/>', () => {
+  const component = shallow(<MenuSelection />);
```
Comment (strange-developer) [inline src/pages/landing-page/menu-selection/component.test.jsx]: Change variable name to wrapper

File: src/pages/landing-page/menu-selection/component.test.jsx [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+
+  it('check menu selection header', () => {
+    const menuHeader = component
```
Comment (strange-developer) [inline src/pages/landing-page/menu-selection/component.test.jsx]: Prefer QA tags

File: src/pages/play-against-computer/component.css [hunk 6]
```diff
@@ -0,0 +1,34 @@
+.game {
+  display: flex;
+  margin-top: 5%;
+  justify-content: center;
+}
+.game-board {
+  margin-top: 3%;
+  height: 30vh;
+  width: 30vw;
```
Comment (strange-developer) [inline src/pages/play-against-computer/component.css:9]: This component will need to be tested on different resolutions

File: src/pages/play-against-computer/component.test.jsx [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+      <PlayAgainstComputer
+        movesForPlayers={movesForPlayers}
+        squares={squares}
+        previousPlayerMoves={previousPlayerMoves}
+      />
+    );
+  });
+
+  it('checks initial game status rendering', () => {
+    const nextPlayer = wrapper
```
Comment (strange-developer) [inline src/pages/play-against-computer/component.test.jsx:35]: Prefer QA tags

File: src/pages/play-against-computer/controller.js [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+};
+
+export const determineGameStatus = (winner, moveNumber, xIsNext) => {
+  if (winner) {
+    return `${winner} is the Winner!`;
+  }
+  if (!winner && moveNumber >= 5) {
+    return 'Its a Draw!';
+  }
+  return `Next player is ${xIsNext ? 'X' : 'O'}`;
```
Comment (strange-developer) [inline src/pages/play-against-computer/controller.js]: Turn this into an `if...else if...else`

File: src/pages/play-against-computer/controller.test.js [hunk 9]
```diff
@@ -0,0 +1,125 @@
...
+});
+
+describe('getPlayerMoves', () => {
+  it('checks moves played by both computer & user', () => {
+    const squareIndex = 3;
+    const xIsNext = true;
+    const history = [
+      Array(9).fill(''),
+      ['X', '', '', '', '', '', '', '', ''],
+
```
Comment (strange-developer) [inline src/pages/play-against-computer/controller.test.js]: No need to leave lines in between the array elements

File: src/reducers/state-of-game/reducer.test.js [hunk 10]
```diff
@@ -8,19 +8,69 @@ describe('reducer', () => {
...
+      ],
+      xIsNext: false,
+      moveNumber: 2,
+    };
+    const expected = {
+      history: [
+        Array(9).fill(''),
+        ['X', '', '', '', '', '', '', '', ''],
+        ['X', '', '', '', '', 'O', '', '', ''],
+        undefined,
```
Comment (strange-developer) [inline src/reducers/state-of-game/reducer.test.js]: Is `undefined` in the expectation correct?

File: src/routes/component.test.jsx [hunk 11]
```diff
@@ -0,0 +1,19 @@
...
+import { LANDING_PAGE } from './urls';
+
+describe('<NavigationRoutes/>', () => {
+  it('checks if route navigates to landing page', () => {
+    const wrapper = shallow(
+      <MemoryRouter initialEntries={[LANDING_PAGE]}>
+        <NavigationRoutes />
+      </MemoryRouter>
+    );
+    expect(wrapper.find(LandingPage)).toHaveLength(0);
```
Comment (strange-developer) [inline src/routes/component.test.jsx]: This doesn't look like a correct test. We can rather test that each route exists.

**Review by strange-developer — CHANGES_REQUESTED** [review]

File: src/reducers/state-of-game/reducer.test.js (hunk 10 above)
Comment (hyperxtend) [reply src/reducers/state-of-game/reducer.test.js]: Yes cause the last array of the current moves isn't added to history until the next move is played

File: src/pages/play-against-computer/component.test.jsx (hunk 7 above)
Comment (strange-developer) [reply src/pages/play-against-computer/component.test.jsx:35]: Here

File: src/reducers/state-of-game/reducer.test.js (hunk 10 above)
Comment (hyperxtend) [reply src/reducers/state-of-game/reducer.test.js]: Updated these tests

**Review by strange-developer — APPROVED** [review]

**Pull request merged**